Tauron amiplus 3 tariff variant 1

darmach opened the pull request:
Adds support for 3 tariff variant of AMIPLUS meter.  This version adds 6 new entities - consumption and production per each of 3 tariffs. Competes with [variant 2](https://github.com/wmbusmeters/wmbusmeters-ha-addon/pull/55)

Closes wmbusmeters/wmbusmeters#1509

weetmuts commented:
Do you know if your changes work as expected? I am not an expert on these files.

darmach commented:
> Do you know if your changes work as expected? I am not an expert on these files.

Unfortunately neither am I! I was hoping someone would be kind enough to help me test these :)
Not sure how to even approach that - should I install wmbusmeters from my fork? Will this break my sensors in HA?
So many questions! :D

darmach commented:
@weetmuts I'm going to test it in the next few days, set up a dev instance of HA with RTL-SDR :)

darmach commented:
@weetmuts Unfortunetely I failed miserably with that - would you have some advice on how to test a change from my fork on new (fresh) homeassistant instance? I set up a separate physical device (rpi) with another RTL dongle and a fresh HA deployment.

darmach commented:
I managed to install my fork to test, doesn't seem to work right - the mqtt autodiscovered device still does not show "tariff" energy entitties...

Wmbusmeters decodes that (debug logging)

```
(shell) env "METER_NAME=test meter"
(shell) env "METER_TYPE=amiplus"
(shell) env "METER_DRIVER="
(shell) env "METER_JSON={"_":"telegram","media":"electricity","meter":"amiplus","name":"test meter","id":"00070787","current_power_consumption_kw":0.021,"current_power_production_kw":0.004,"total_energy_consumption_tariff_1_kwh":2546.412,"total_energy_consumption_tariff_2_kwh":0,"total_energy_consumption_tariff_3_kwh":0,"total_energy_production_tariff_1_kwh":369.661,"total_energy_production_tariff_2_kwh":0,"total_energy_production_tariff_3_kwh":0,"voltage_at_phase_1_v":237,"voltage_at_phase_2_v":241,"voltage_at_phase_3_v":242,"device_date_time":"2025-04-01 18:20:11","timestamp":"2025-04-01T18:20:08Z","device":"rtlwmbus[00000001]","rssi_dbm":42}"
(shell) env "METER_MEDIA=electricity"
(shell) env "METER_TIMESTAMP=2025-04-01T18:20:08Z"
(shell) env "METER_TIMESTAMP_UTC=2025-04-01T18:20:08Z"
(shell) env "METER_TIMESTAMP_UT=1743531608"
(shell) env "METER_TIMESTAMP_LT=2025-04-01 20:20:08"
(shell) env "METER_TOTAL_ENERGY_CONSUMPTION_KWH=null"
(shell) env "METER_CURRENT_POWER_CONSUMPTION_KW=0.021"
(shell) env "METER_TOTAL_ENERGY_PRODUCTION_KWH=null"
(shell) env "METER_CURRENT_POWER_PRODUCTION_KW=0.004"
(shell) env "METER_VOLTAGE_AT_PHASE_1_Volt=237"
(shell) env "METER_VOLTAGE_AT_PHASE_2_Volt=241"
(shell) env "METER_VOLTAGE_AT_PHASE_3_Volt=242"
(shell) env "METER_DEVICE_DATE_TIME=2025-04-01 18:20:11"
(shell) env "METER_TOTAL_ENERGY_CONSUMPTION_TARIFF_1_KWH=2546.412"
(shell) env "METER_TOTAL_ENERGY_CONSUMPTION_TARIFF_2_KWH=0"
(shell) env "METER_TOTAL_ENERGY_CONSUMPTION_TARIFF_3_KWH=0"
(shell) env "METER_TOTAL_ENERGY_PRODUCTION_TARIFF_1_KWH=369.661"
(shell) env "METER_TOTAL_ENERGY_PRODUCTION_TARIFF_2_KWH=0"
(shell) env "METER_TOTAL_ENERGY_PRODUCTION_TARIFF_3_KWH=0"
(shell) env "METER_MAX_POWER_CONSUMPTION_KW=null"
(shell) env "METER_DEVICE=rtlwmbus[00000001]"
(shell) env "METER_RSSI_DBM=42"
(shell) env "METER_FIRST_TELEGRAM=true"
```

As you can see, it seem that in my meters firmware (?) it supports 3 tariffs (despite I'm having only one in the contract) and as such - all the energy is billed under Tariff 1. I attempted to add tariffs 1 2 3 as an alternative to currently present in the driver json `total_energy_consumption_kwh`. I think I had misconception on how this part of the code works.

Mice30 commented:
![image](https://github.com/user-attachments/assets/fe669326-c6e9-4ed5-9320-70e1c4eab811)
[amiplus_json_enchenced.txt](https://github.com/user-attachments/files/19605351/amiplus_json_enchenced.txt)

This is a version witch works for me.

darmach commented:
@Mice30 Thanks! Not sure what I was doing wrong - installed the addon from my fork, where I merged these changes. Now I just added samba addon, and modified amiplus.json locally and this did the trick. It shows new entities working, the old ones however are showing as Unknown.

Do you  know if there is anyway to hide them if unknown? I'd prefer not to remove anything, as I suspect it would break things for people with amiplus meter not having different tariffs support.

Mice30 commented:
@darmach Can you post screenshots?

Mice30 commented:
![image](https://github.com/user-attachments/assets/9addfde3-931c-4ec8-82ea-0fb91ec80da0)
Restart Wmbusmeters and see at logs whats there.

darmach commented:
@Mice30 I know why I have these two unknowns, my point is, that I don't want to remove them from the codebase. As they were already in amiplus.json when I started my change, I suppose they are used by some variant of amiplus meter. That is a safe assumption, right?

```
[16:09:07] INFO:  Adding meter: Amiplus Test ...
[16:09:08] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_current_power_consumption_kw/config
[16:09:09] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_current_power_production_kw/config
[16:09:10] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_device_date_time/config
[16:09:11] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_rssi_dbm/config
[16:09:12] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_timestamp/config
[16:09:12] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_total_energy_consumption_kwh/config
[16:09:13] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_total_energy_consumption_tariff_1_kwh/config
[16:09:14] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_total_energy_consumption_tariff_2_kwh/config
[16:09:15] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_total_energy_consumption_tariff_3_kwh/config
[16:09:16] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_total_energy_production_kwh/config
[16:09:16] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_total_energy_production_tariff_1_kwh/config
[16:09:17] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_total_energy_production_tariff_2_kwh/config
[16:09:18] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_total_energy_production_tariff_3_kwh/config
[16:09:19] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_voltage_at_phase_1_v/config
[16:09:19] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_voltage_at_phase_2_v/config
[16:09:20] INFO:   Add/update topic: homeassistant/sensor/wmbusmeters/56755731_voltage_at_phase_3_v/config
[16:09:20] INFO: MQTT Discovery cleanup...
[16:09:21] INFO: Running wmbusmeters ...
```

darmach commented:
In other PR I attempted to template a value:

https://github.com/wmbusmeters/wmbusmeters-ha-addon/pull/55/files#diff-e8668865aa0846bdf901661183d109d31648d3fd10e1ef3819c6e42d63fc4fa2L17-R17

I'll give it a try and report back!

darmach commented:
@Mice30 Ha! It works! 

![image](https://github.com/user-attachments/assets/6a76803c-e6a7-45bd-afca-fbc6288c41b7)

So:
1. I can template the value for `total energy consumption` conditionally - to use non-tariff data when present, or tariff when not. This will allow `total energy consumption` to present proper data (sum) for both use cases.
2. I still have ALL of them (8!) `total energy consumption/production (tariff 1/2/3)` posted to HA. Is there a way to make them "disabled" if `Unknown? (They was voltages or device timestamp are marked as disabled by default - pic below)

![image](https://github.com/user-attachments/assets/b49a887f-0bff-4876-bafe-2905871ae5f9)

Mice30 commented:
@darmach Yeah, we should think about backward compatibility as much as possible :) Tauron loves to surprise us—even in this area! :D

I'll try to integrate your work tomorrow (if you're willing to share it :D).

darmach commented:
Pushed an update to the fork branch. Still would like to figure out how to "hide" (disable?) empty values, but for now this should work fine.

darmach commented:
@weetmuts Verified these changes and confirmed working, would you take a look? I would like to "hide" empty/unknown values, but we can merge this one, and I'll open another when I figure out how :)

darmach commented:
> > Try GPTit:)

Good point, those damn things are awfully good at times!

darmach commented:
> Do you know if your changes work as expected? I am not an expert on these files.

Confirmed working @weetmuts ! :)

SandisOU commented:
Tnx @darmach, but please move file to edge directory.

BIBOLV commented:
Tnx @darmach, but please make changes to edge directory.

darmach commented:
@BIBOLV Rebased those 3 restoring original "prod" version of amiplus.json, and patching "edge".

BIBOLV commented:
Thanks!
